### PR TITLE
CodeJam-3 CoreJS

### PR DESCRIPTION
**Pull request by kalitamih**
This is a CodeJam about CoreJS. This  consists of 3 tasks: array methods, closure and recursion. I did my best to solve all of it.

### CODE REVIEW

File: make/make.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = function make(...args) {
+  const temporyArray = [];
+  for (let i = 0; i < args.length; i += 1) temporyArray.push(args[i]);
+  function inner(...rest) {
+    if (typeof rest[0] === 'function') return temporyArray.reduce(rest[0]);
+    for (let i = 0; i < rest.length; i += 1) temporyArray.push(rest[i]);
```
Comment (NikitaRudy) [inline make/make.js:6]: instead you could use
```js
temporyArray.push(...rest);
```

File: make/make.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = function make(...args) {
+  const temporyArray = [];
```
Comment (NikitaRudy) [inline make/make.js:2]: I think `temporyArray` isn't appropriate variable name here.

File: make/test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+function sum(a, b) {
+  return a + b;
+}
+
+function multiply(a, b) {
+  return a * b;
+}
+
+describe('make', () => {
+  it('1', () => {
```
Comment (NikitaRudy) [inline make/test.js:13]: It block should contain some description about test case.
For example:
`Should handle partial application of arguments`
`Should invoke function with memoized args when it is provided`
etc.

File: make/test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    const check = make(15)(34, 21, 666)(41)(sum);
+    assert.deepEqual(check, 777);
+  });
+  it('2', () => {
```
Comment (NikitaRudy) [inline make/test.js:17]: All your `it` blocks tests similar behavior and should be combined in one `it`.
Also you should add some edge case testing and failure testing if rules of function isn't followed.

File: recursion/test.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+const tree3 = { value: 100 };
+
+const tree4 = {
+  value: 100,
+  left: { value: 87, left: { value: 70 }, right: { value: 90 } },
+  right: { value: 120, left: { value: 110 }, right: { value: 1900 } },
+};
+
+describe('recursion', () => {
```
Comment (NikitaRudy) [inline recursion/test.js:24]: same comments as for previous test for make.js

File: sumOfOther/test.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+const assert = require('assert');
+const sumOfOther = require('./sumOfOther.js');
+
+describe('sumOfOther', () => {
+  it('1', () => {
```
Comment (NikitaRudy) [inline sumOfOther/test.js:5]: This function is pretty strait forward, therefore it is difficult to create some edge case tests.

But again, you could combine all `it` blocks into one and name it properly. Also you could add test for empty array, or for undefined and etc.

**Review by NikitaRudy — APPROVED** [review]
The score is 120
Great job!
